isyfact-standards-doc/src/docs/antora/modules/referenzarchitektur/pages/software-technisch/backend/spring.adoc

```diff
@@ -0,0 +1,290 @@
+@ComponentScan
+public class CoreConfig {
+
+    @Bean
+    public BeispielBean beispielBean() {
+        // ...
+    }
+}
+----
+
+Die Spring-Beans der Komponenten werden mit der Annotation `@Component` bzw. deren Spezialisierungen (`@Repository`, `@Service`, `@Controller` etc.) versehen.
+Externe Spring-Beans, die nicht annotiert werden können, werden in der Konfigurationsklasse der Schicht per `@Bean`-Methoden konfiguriert.
+
+[[webapprootkey-konfigurieren]]
+=== webAppRootKey konfigurieren
+
+Der Kontextparameter `webAppRootKey` muss in der `web.xml` auf den eindeutigen Namen der Webanwendung gesetzt werden.
+
+[[listing-webAppRootKey]]
+.Konfiguration des webAppRootKey
+[source,xml]
+----
+<context-param>
+  <param-name>webAppRootKey</param-name>
+  <param-value>allgemeine-anwendung</param-value>
+</context-param>
+----
+
+*Hintergrund*: Spring speichert standardmäßig den Pfad zum Wurzelverzeichnis der Webanwendung im Webserver in der Kontextvariable `webapp.root`.
+Wenn mehrere Anwendungen gleichzeitig in einem Tomcat betrieben werden (bspw.
+in einer Entwicklungsumgebung), wird dieser Parameter durch die verschiedenen Anwendungen überschrieben.
+Dies kann zu ungewünschten Seiteneffekten führen.
+Ist der Kontextparameter `webAppRootKey` wie im obigen Beispiel gesetzt, wird der Pfad statt im Parameter `webapp.root`
+im Parameter `allgemeine-anwendung` abgelegt.
+Da jede Webanwendung einen eindeutigen Namen besitzt, und damit einen eigenen Kontextparameter verwendet, wird das Überschreiben vermieden.
+
+[[direkte-verwendung-des-spring-frameworks]]
+== Direkte Verwendung von Spring
+
+Neben der Konfiguration der Spring-Beans enthält eine Anwendung kaum Abhängigkeiten zu Spring.
+Da die benötigten Objekte per Dependency Injection konfiguriert werden, müssen diese nach ihrer Konfiguration lediglich verwendet werden.
+
+In einigen Fällen existieren jedoch weitere Abhängigkeiten:
+
+* Spring-Beans werden per Namen aus dem Anwendungskontext ausgelesen.
+* Spring-Beans werden nicht per Dependency Injection, sondern über statische Methoden bereitgestellt.
+Hierüber entstehen Abhängigkeiten zur Instanziierungsreihenfolge der Spring-Beans durch Spring.
+
+Für die Themen werden im folgenden Vorgaben aufgestellt.
+
+[[keine-beans-per-namen-auslesen]]
+=== Keine Spring-Beans per Namen auslesen
+
+Über den Anwendungskontext könnten Spring-Beans explizit per Namen ausgelesen werden.
+Dies ist mit einer Ausnahme verboten: Die Namen von Spring-Beans sollen nicht im Anwendungscode verwendet werden.
+Die Ausnahme gilt für den Zugriff von einem Anwendungskontext auf einen anderen (in Zusammenhang mit dem DispatcherServlet).
+In diesem Fall ist ein explizites Auslesen nicht zu vermeiden.
+Auszulesen ist in diesem Fall keine Komponente des Anwendungskerns, sondern eine weitere Schnittstellen-Bean, welche nur für diesen Zweck verwendet wird.
+
+[[aspektorientierte-programmierung-in-spring]]
+== Aspektorientierte Programmierung in Spring
```

Comment:
Geht es in diesem Kapitel um Vorgaben?
Wenn ja, dann sollte diese Kapitel in "Vorgaben zur aspektorierten Programmierung in Spring" umbenenenn